File: algorithms/rosalind/lexf.rb

def lexo_strings(symbols, n)
  strings = symbols

  while n > 1
    strings = strings.reduce([]) do |new_strings, string|
      new_strings.concat(symbols.map { |sym| string + sym  })
    end
    n -= 1
  end

  return strings
end

syms = "R Z O B A L Y".split(' ')
n = 3

puts lexo_strings(syms, n)

File: exercises/rosalind/lexf.clj

(defn lexo_strings [symbols n]
  (loop [strings symbols c 1]
    (if (= c n)
      strings
      (recur
        (reduce (fn [col string]
                  (concat col (map (partial str string) symbols))) 
                [] strings)
        (inc c)))))


(def syms ["T" "A" "G" "C"])
(def n 2)

(println (lexo_strings syms n))
